notebooks/problem-sets/quantum_enigma_problem_set.ipynb

```diff
@@ -0,0 +1,381 @@
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-quiz-0\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "Ready for your quiz?\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. Yes -- Let's go! \n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. Not yet, I need to watch the video again.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 1\n",
+    "\n",
+    "Sometimes a quantum circuit can be simplified. One way of achieving this is by cancelling some quantum gates. Could you simplify the following circuit?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Click run to draw the circuit\n",
+    "\n",
+    "from qiskit import QuantumCircuit\n",
+    "\n",
+    "problem_qc = QuantumCircuit(3)\n",
+    "\n",
+    "problem_qc.h(0)\n",
+    "problem_qc.h(2)\n",
+    "problem_qc.cx(0, 1)\n",
+    "problem_qc.barrier(0, 1, 2)\n",
+    "problem_qc.cx(2, 1)\n",
+    "problem_qc.x(2)\n",
+    "problem_qc.cx(2, 0)\n",
+    "problem_qc.x(2)\n",
+    "problem_qc.barrier(0, 1, 2)\n",
+    "problem_qc.swap(0, 1)\n",
+    "problem_qc.x(1)\n",
+    "problem_qc.cx(2, 1)\n",
+    "problem_qc.x(0)\n",
+    "problem_qc.x(2)\n",
+    "problem_qc.cx(2, 0)\n",
+    "problem_qc.x(2)\n",
+    "\n",
+    "problem_qc.draw()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Try simplifying the circuit and rerun the calculation between each simplification to make sure you always get the same histogram. Check your answer by clicking the `Grade` button\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    The NOT, CNOT, and Hadamard gates are their own inverse. That means that if two of these gates are placed side by side they can simply be taken off.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    The SWAP gate can be taken off if the subsequent operations are adjusted between the two qubits.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 3</summary>\n",
+    "    If a CNOT has the same control and target as another CNOT for which two NOT gates are applied before and after the control qubit, this can be simplified to a single NOT gate on the target qubit of the CNOT as a NOT gate is applied to the target whether the control qubit is initially in state 0 or 1.\n",
```

Comment:
Add the following picture after the explanation of Hint 3

![Identite](https://user-images.githubusercontent.com/64601645/206603983-d59437c6-525b-4bac-b13c-58800ce4f459.png)